tests/Normalization.hs

```diff
@@ -29,6 +31,36 @@ conversions = testGroup "conversions" [ naturalShow
                                       , naturalToInteger
                                       ]
 
+customization :: TestTree
+customization = testGroup "customization"
+                 [simpleCustomization
+                 ,doubleReduction]
+
+simpleCustomization :: TestTree
+simpleCustomization = testCase "simpleCustomization" $ do
+  let tyCtx  = insert "min" (Pi "_" Natural (Pi "_" Natural Natural)) empty 
+      valCtx e = case e of
+                    (App (App (Var (V "min" 0)) (NaturalLit x)) (NaturalLit y)) -> Just (NaturalLit (min x y))
+                    _ -> Nothing
+  e <- codeWith tyCtx "min (min +11 +12) +8 + +1" 
+  assertNormalizesToWith valCtx e "+9"
+
+doubleReduction :: TestTree
+doubleReduction = testCase "doubleReduction" $ do
+  let tyCtx  =  insert "min" (Pi "_" Natural (Pi "_" Natural Natural)) 
```

Comment:
Minor suggestion: you can use `code` to input these types:

```haskell
minType        <- code "Natural → Natural → Natural"
fiveorlessType <- code "Natural → Natural"
wurbleType     <- code "Natural → Integer
let tyCtx = insert "min"        minType
          . insert "fiveorless" fiveorlessType
          . insert "wurble"     wurbleType
          $ empty
```

I don't feel very strongly about this suggestion, so feel free to ignore me :)